web_diagram_position/static/src/js/graph.js

```diff
@@ -0,0 +1,1075 @@
+                    'stroke': graph.style.edge_color,
+                    'stroke-width': graph.style.edge_width
+                }, 100, 'linear');
+                edge_label.animate({'fill': graph.style.edge_label_color}, 100, 'linear');
+                if(self.close_button){
+                    self.close_button.remove();
+                    self.close_button = null;
+                }
+            }
+        };
+
+        this.set_label = function(text){
+            edge_label.attr({'text': text});
+        };
+
+        function click_action(){
+            graph.select(self);
+        }
+
+        if (!graph.is_readonly()) {
+            edge.click(click_action);
+            edge_label.click(click_action);
+        }
+
+        function double_click_action(){
+            MyGraphEdge.double_click_callback(self);
+        }
+
+        if (!graph.is_readonly()) {
+            edge.dblclick(double_click_action);
+            edge_label.dblclick(double_click_action);
+        }
+
+
+        this.label_enable  = label_enable;
+        this.label_disable = label_disable;
+        this.update = update;
+
+        if (!graph.is_readonly()) {
+            this.remove = remove;
+        }
+
+        this.is_loop = function(){ return start === end; };
+        this.get_start = function(){ return start; };
+        this.get_end   = function(){ return end; };
+    }
+
+    MyGraphEdge.double_click_callback = function(edge){
+        console.log("double click from edge:",edge);
+    };
+
+    // this is the default edge creation callback. It is called before an edge is created
+    // It returns an object containing the properties of the edge.
+    // If it returns null, the edge is not created.
+    MyGraphEdge.creation_callback = function(start, end){
+        var edge_prop = {};
+        edge_prop.label = 'new edge!';
+        return edge_prop;
+    };
+    // This is is called after a new edge is created, with the new edge
+    // as parameter
+    MyGraphEdge.new_edge_callback = function(new_edge){};
+
+    // this is the default edge destruction callback. It is called before
+    // an edge is removed from the graph.
+    MyGraphEdge.destruction_callback = function(edge){ return true; };
+
+    // returns a new string with the same content as str, but with lines of maximum 'width' characters.
+    // lines are broken on words, or into words if a word is longer than 'width'
+    function wordwrap( str, width) {
+        // http://james.padolsey.com/javascript/wordwrap-for-javascript/
+        width = width || 32;
+        var cut = true;
+        var brk = '\n';
+        if (!str) { return str; }
+        var regex = '.{1,' +width+ '}(\\s|$)' + (cut ? '|.{' +width+ '}|.+$' : '|\\S+?(\\s|$)');
+        return str.match(new RegExp(regex, 'g') ).join(brk);
+    }
+
+    window.MyCuteGraph = MyGraph;
+    window.MyCuteNode = MyGraphNode;
+    window.MyCuteEdge = MyGraphEdge;
+    window.MyCuteGraph.wordwrap = wordwrap;
+})(window);
```

Comment:
Cant you make changes by overriding this js ? I think there has to be a good way to handle this.